### ISSUE TITLE
run below code and send screenshot of results

### ISSUE THREAD

**Issue by kkiritok** [kkiritok/Quantam-based-optimization#14]
import torch
import torchvision
import torchvision.transforms as transforms
from torchvision.models import vgg16
import random
import copy

# ---------------- DEVICE ----------------
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
print("Using:", device)

# ---------------- DATA ----------------
transform_train = transforms.Compose([
    transforms.Resize(224),
    transforms.RandomHorizontalFlip(),
    transforms.RandomCrop(224, padding=4),
    transforms.ToTensor(),
    transforms.Normalize(
        mean=[0.485, 0.456, 0.406],
        std=[0.229, 0.224, 0.225]
    )
])

transform_test = transforms.Compose([
    transforms.Resize(224),
    transforms.ToTensor(),
    transforms.Normalize(
        mean=[0.485, 0.456, 0.406],
        std=[0.229, 0.224, 0.225]
    )
])

trainset = torchvision.datasets.CIFAR100(
    root='./data', train=True, download=True, transform=transform_train
)

testset = torchvision.datasets.CIFAR100(
    root='./data', train=False, download=True, transform=transform_test
)

trainloader = torch.utils.data.DataLoader(trainset, batch_size=64, shuffle=True)
testloader = torch.utils.data.DataLoader(testset, batch_size=64, shuffle=False)

# ---------------- MODEL ----------------
model = vgg16(weights="IMAGENET1K_V1")

# Replace classifier (100 classes)
model.classifier[6] = torch.nn.Linear(4096, 100)

model = model.to(device)

# ---------------- FREEZE ALL ----------------
for p in model.parameters():
    p.requires_grad = False

# ---------------- ACCURACY ----------------
def accuracy_loader(model, loader):
    model.eval()
    correct, total = 0, 0

    with torch.no_grad():
        for x, y in loader:
            x, y = x.to(device), y.to(device)
            out = model(x)
            _, pred = out.max(1)
            total += y.size(0)
            correct += pred.eq(y).sum().item()

    return correct / total

# ---------------- INITIAL ACC ----------------
best_acc = accuracy_loader(model, testloader)
print("Initial Test Accuracy:", best_acc)

# ---------------- VGG BLOCKS ----------------
# VGG features split into blocks
layers = {
    "block1": model.features[:5],    # conv1
    "block2": model.features[5:10],  # conv2
    "block3": model.features[10:17],
    "block4": model.features[17:24],
    "block5": model.features[24:31]
}

patience = 3
no_improve_count = 0
max_iterations = 40

# ---------------- TRAIN LOOP ----------------
for i in range(max_iterations):

    print("\n========== Iteration", i + 1, "==========")

    chosen_names = random.sample(
        list(layers.keys()),
        random.choice([1, 2])
    )

    print("Unfreezing:", chosen_names)

    old_weights = copy.deepcopy(model.state_dict())

    # -------- UNFREEZE RANDOM BLOCKS --------
    for name in chosen_names:
        for p in layers[name].parameters():
            p.requires_grad = True

    # -------- ALWAYS TRAIN CLASSIFIER --------
    for p in model.classifier.parameters():
        p.requires_grad = True

    optimizer = torch.optim.AdamW(
        filter(lambda p: p.requires_grad, model.parameters()),
        lr=3e-4,
        weight_decay=1e-4
    )

    scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(
        optimizer, T_max=8
    )

    loss_fn = torch.nn.CrossEntropyLoss(label_smoothing=0.1)

    # -------- TRAIN --------
    model.train()

    for epoch in range(8):

        running_loss = 0

        for x, y in trainloader:
            x, y = x.to(device), y.to(device)

            optimizer.zero_grad()
            out = model(x)
            loss = loss_fn(out, y)
            loss.backward()
            optimizer.step()

            running_loss += loss.item()

        scheduler.step()
        print(f"Epoch {epoch+1} Loss: {running_loss/len(trainloader):.4f}")

    # -------- EVAL --------
    train_acc = accuracy_loader(model, trainloader)
    test_acc = accuracy_loader(model, testloader)

    print("Train Accuracy:", train_acc)
    print("Test Accuracy :", test_acc)

    # -------- SAVE BEST --------
    if test_acc > best_acc:

        best_acc = test_acc
        no_improve_count = 0

        torch.save(model.state_dict(), "best_vgg16.pth")

        print("✅ Improved — Model Saved")

    else:

        model.load_state_dict(old_weights)
        no_improve_count += 1

        print("❌ Reverted | No Improve Count:", no_improve_count)

    # -------- FREEZE AGAIN --------
    for p in model.parameters():
        p.requires_grad = False

    if no_improve_count >= patience:
        print("\n🛑 Early stopping triggered!")
        break

print("\nFinal Best Accuracy:", best_acc)
